include/z64thread.h

```diff
@@ -0,0 +1,24 @@
+#ifndef Z64THREAD_H
+#define Z64THREAD_H
+
```

Comment:
More so just a thought. Perhaps these would be better off in their individual headers where most of their thread structs live (for example Z_THREAD_ID_SLOWLY and Z_PRIORITY_SLOWLY in `slowly.h`). This doesn't quite work for Main and Idle since they don't have their own headers, but I think the rest do. However I also see the value of having them all together to easily compare, not sure 🤷.

Reply:
At that point we could just have the defines in each C file instead, but I kinda like having them grouped together in the same place. Those could even be an enum but I don't know if we really want that